### PR TITLE
SCIP: populate new SymbolInformation fields

### LINKED ISSUE #15919: Implement new `SymbolInformation` fields in the SCIP backend
sourcegraph/scip/issues/154 added support for a number of new fields in SCIP's `SymbolInformation`:
 * `kind`
 * `display_name`
 * `signature_documentation`
 * `enclosing_symbol` (what was originally proposed as `owner` or `parent` in the linked issue)

rust-lang/rust-analyzer/pull/15806 recently updated scip to 0.3.1 which makes those new fields available to rust-analyzer. They are currently always defaulted.

@donsbot you said you wanted to add support for those in rust-analyzer, do you still plan on doing that?
If not I'm also interested in working on that, for use in Searchfox.

### PR DESCRIPTION
**Pull request by nicolas-guichard**
Fixes #15919

### CODE REVIEW

File: crates/ide-db/src/defs.rs [hunk 1]
```diff
@@ -134,6 +143,132 @@ impl Definition {
...
+            let name = Some(assoc.name(db)?);
+            let item = trait_.items(db).into_iter().find(|it| it.name(db) == name)?;
+            item.docs(db)
+        })
+    }
+
+    pub fn label(
+        &self,
+        db: &RootDatabase,
+        famous_defs: Option<&FamousDefs<'_, '_>>,
```
Comment (Veykril) [inline crates/ide-db/src/defs.rs:213]: This can be removed fully, its only used for its `db` field but the `db` is already being passed here

**Comment by Veykril** [comment]
One small thing, but other lgtm!

File: crates/ide-db/src/defs.rs (hunk 1 above)
Comment (nicolas-guichard) [reply crates/ide-db/src/defs.rs]: Indeed, I removed it.

**Comment by nicolas-guichard** [comment]
In addition to the change requested above, I've also rebased and added some tests for TraitAlias and TypeAlias which were not previously covered. I think the existing behavior for plain type aliases was unexpected (`type MyTypeAlias = u8;` yielded `[MyTypeAlias]` instead of `MyTypeAlias#`), but do tell me if you would rather revert to that.

**Review by Veykril — APPROVED** [review]

**Comment by Veykril** [comment]
Thanks!
@bors r+

**Pull request merged**